lambdas/journeyengine/src/main/java/uk/gov/di/ipv/core/journeyengine/JourneyEngineHandler.java

```diff
@@ -147,33 +140,36 @@ private JourneyEngineResult executeJourneyEvent(
                             new JourneyResponse(criStartUri + UK_PASSPORT_CRI_ID));
                     break;
                 case CRI_UK_PASSPORT:
-                    if (journeyStep.equals(NEXT_STEP)) {
+                    if (journeyStep.equals(JourneyStep.NEXT.toString())) {
                         updateUserState(CRI_ADDRESS, ipvSessionItem);
                         builder.setJourneyResponse(
                                 new JourneyResponse(criStartUri + ADDRESS_CRI_ID));
-                    } else if (journeyStep.equals(ERROR_STEP)) {
+                    } else if (journeyStep.equals(JourneyStep.ERROR.toString())) {
+                        updateUserState(CRI_ERROR, ipvSessionItem);
+                        builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
+                    } else if (journeyStep.equals(JourneyStep.FAIL.toString())) {
                         updateUserState(CRI_ERROR, ipvSessionItem);
                         builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
                     } else {
                         handleInvalidJourneyStep(journeyStep, CRI_UK_PASSPORT.value);
                     }
                     break;
                 case CRI_ADDRESS:
-                    if (journeyStep.equals(NEXT_STEP)) {
+                    if (journeyStep.equals(JourneyStep.NEXT.toString())) {
                         updateUserState(CRI_FRAUD, ipvSessionItem);
                         builder.setJourneyResponse(new JourneyResponse(criStartUri + FRAUD_CRI_ID));
-                    } else if (journeyStep.equals(ERROR_STEP)) {
+                    } else if (journeyStep.equals(JourneyStep.ERROR.toString())) {
                         updateUserState(CRI_ERROR, ipvSessionItem);
                         builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
                     } else {
                         handleInvalidJourneyStep(journeyStep, CRI_ADDRESS.value);
                     }
                     break;
                 case CRI_FRAUD:
-                    if (journeyStep.equals(NEXT_STEP)) {
+                    if (journeyStep.equals(JourneyStep.NEXT.toString())) {
                         updateUserState(PRE_KBV_TRANSITION_PAGE, ipvSessionItem);
                         builder.setPageResponse(new PageResponse(PRE_KBV_TRANSITION_PAGE.value));
-                    } else if (journeyStep.equals(ERROR_STEP)) {
+                    } else if (journeyStep.equals(JourneyStep.ERROR.toString())) {
```

Comment:
It's probably worth adding the fail branch for the fraud CRI. It'll need to redirect to the same error page as the passport CRI fail scenario.

Reply:
Done